go/buildhelper/helpertypes/types.go

```diff
@@ -0,0 +1,130 @@
+package helpertypes
+
+import (
+	"bytes"
+	"compress/gzip"
+	b64 "encoding/base64"
+	"io/ioutil"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/obscuronet/obscuro-playground/go/buildhelper/buildconstants"
+	"github.com/obscuronet/obscuro-playground/go/obscurocommon"
+)
+
+var IsRealEth bool
+
+func UnpackL1Tx(tx *types.Transaction) *obscurocommon.L1TxData {
+	if !IsRealEth {
+		t := obscurocommon.TxData(tx)
+		return &t
+	}
+	// ignore transactions that are not calling the contract
+	if tx.To() == nil || tx.To().Hex() != buildconstants.CONTRACT_ADDRESS.Hex() || len(tx.Data()) == 0 {
+		return nil
+	}
+
+	contractABI, err := abi.JSON(strings.NewReader(buildconstants.CONTRACT_ABI))
+	if err != nil {
+		panic(err)
+	}
+
+	//log.Log(fmt.Sprintf("Unpacking data: %b", tx.Data()))
+	method, err := contractABI.MethodById(tx.Data()[:4])
+	if err != nil {
+		panic(err)
+	}
+
+	l1txData := obscurocommon.L1TxData{
+		TxType:      0,
+		Rollup:      nil,
+		Secret:      nil,
+		Attestation: obscurocommon.AttestationReport{},
+		Amount:      0,
+		Dest:        common.Address{},
+	}
+	switch method.Name {
+	case "Deposit":
+		contractCallData := map[string]interface{}{}
```

Comment:
@otherview You can pull the repeated first two lines of each `case` to outside of the `switch` statement.

Reply:
Yep 👍